### DISCUSSION TITLE
.resw Localization can't correctly work

### DISCUSSION THREAD

**Discussion by DJDQfff** [unoplatform/uno#13488] [Q&A] [answered]
I want to use the localization words with .resw files, but met an error.

In XAML: 
```xaml
    <StackPanel Orientation="Vertical">
        <TextBlock Loaded="TextBlock_Loaded" />
        <TextBlock x:Uid="String2" />
        <TextBlock x:Uid="/Resources/String2"/>
    </StackPanel>
```
Platforms:
* WinUI: the first, third TextBlock can't work.
* WASM: the first, TextBlock can't work.

In XAML: 
```xaml
    <StackPanel Orientation="Vertical">
        <TextBlock Loaded="TextBlock_Loaded" />
        <TextBlock x:Uid="String2" />
<TextBlock x:Uid="/Resources/String2"/>
    </StackPanel>
```
In he first TextBlock can' t work , the methos `Windows.ApplicationModel.Resources.ResourceLoader.GetForViewIndependentUse().GetString("String1")`  return an empty string.

The second TextBlock can work .

The third TextBlock can't work. I added more .resw files, use ` x:Uid=/Resources2/string3` also doesn't work

![图片](https://github.com/unoplatform/uno/assets/31266256/fae5911b-b8f1-43a4-80bc-bf8aaefbd5aa)
![图片](https://github.com/unoplatform/uno/assets/31266256/b61d1665-f6da-4e00-a714-b9fcf5be7696)

**Comment by jeromelaban** [comment]
Thanks for the report. Could you attach a repro sample that shows what you're trying to do? Thanks!

**Reply by DJDQfff** [reply]
https://github.com/DJDQfff/LocializationTest.git

**Reply by jeromelaban** [reply, marked as answer]
Thanks. Localization from XAML from a separate project file using a non-default resource file name requires prefixing with the `x:Uid` with the assembly name:

```xml
<TextBlock x:Uid="/LocializationTest/Resources1/String1" />
```

**Comment by KWodarczyk** [comment]
Is it possible to use same resource for different elements e.g Button and TextBlock ? Since i have to add a property to a Resource in one case it will be TestString.Content and other TestString.Text  would be nice if it worked without providing properties e.g just "TestString"

**Reply by jeromelaban** [reply]
This is not a behavior that WinUI provides. The names of the resources need to match the property name they're bound to.
